tests/tracer/test_span.py

```diff
@@ -561,3 +561,16 @@ def test_on_finish_multi_callback():
     s.finish()
     m1.assert_called_once_with(s)
     m2.assert_called_once_with(s)
+
+
+def test_span_pprint():
+    s = Span(None, "test")
+    s.set_tag("foo", "bar")
+    s.set_metric("asd", 42)
+    sp = s.pprint()
+
+    # TODO: These checks are weak
```

Comment:
We _should_ be able to assert on specific output string?

There are a few properties which are dynamic, but maybe a regex would work?

It is a tricky one

Reply:
It's definitely possible, but the fact that there was no test case for `pprint` in the first place made me think that it's perhaps not worth spending time to add quite some complexity to test this thoroughly. But I'm happy to give it a go 🙂

Reply:
`pprint` is likely only ever called if we are running with debug logging. on every trace finish we `pprint` all the spans in the trace.